packages/shared/src/components/plus/GiftPlusModal.tsx

```diff
@@ -0,0 +1,221 @@
+        }
+        return next % users.length;
+      });
+    } else if (e.key === 'ArrowUp') {
+      setIndex((prev) => {
+        let next = prev - 1;
+        while (users[(next + users.length) % users.length]?.isPlus) {
+          next -= 1;
+        }
+        return (next + users.length) % users.length;
+      });
+    } else {
+      setSelected(users[index]);
+    }
+  };
+
+  const onSelect = (user: UserShortProfile) => {
+    setSelected(user);
+    setIndex(0);
+    setQuery('');
+  };
+
+  return (
+    <Modal
+      {...props}
+      isOpen
+      kind={Modal.Kind.FixedCenter}
+      size={Modal.Size.Small}
+      overlayRef={setOverlay}
+    >
+      <Modal.Body className="gap-4">
+        <div className="flex flex-row justify-between">
+          <PlusTitle type={TypographyType.Callout} bold />
+          <CloseButton
+            type="button"
+            size={ButtonSize.Small}
+            onClick={onRequestClose}
+          />
+        </div>
+        <Typography bold type={TypographyType.Title1}>
+          Gift daily.dev Plus 🎁
+        </Typography>
+        {selected ? (
+          <SelectedUser user={selected} onClose={() => setSelected(null)} />
+        ) : (
+          <div className="flex flex-col">
+            <BaseTooltip
+              appendTo={overlay}
+              onClickOutside={() => setQuery('')}
+              visible={isVisible}
+              showArrow={false}
+              interactive
+              content={
+                <RecommendedMention
+                  users={users}
+                  selected={index}
+                  onClick={onSelect}
+                  onHover={setIndex}
+                  checkIsDisabled={(user) => user.isPlus}
+                />
+              }
+              container={{
+                className: 'shadow',
+                paddingClassName: 'p-0',
+                roundedClassName: 'rounded-16',
+                bgClassName: 'bg-accent-pepper-subtlest',
+              }}
+            >
+              <TextField
+                leftIcon={<UserIcon />}
+                inputId="search_user"
+                fieldType="tertiary"
+                autoComplete="off"
+                label="Select a recipient by name or handle"
+                onKeyDown={onKeyDown}
+                onChange={(e) => onSearch(e.currentTarget.value.trim())}
+                onFocus={(e) => setQuery(e.currentTarget.value.trim())}
+              />
+            </BaseTooltip>
+          </div>
+        )}
+        <div className="flex w-full flex-row items-center gap-2 rounded-10 bg-surface-float p-2">
+          <Typography bold type={TypographyType.Callout}>
+            One-year plan
+          </Typography>
+          <Typography
+            bold
+            className="rounded-10 bg-action-upvote-float px-2 py-1"
+            type={TypographyType.Caption1}
+            color={TypographyColor.StatusSuccess}
+          >
+            2 months free
+          </Typography>
```

Comment:
This should come from the data from Paddle no?

Reply:
Right. Although the one-time payment details will be adjusted from a different task, I will now make this label be coming from the object by paddle so it won't be forgotten.

Do you know which property this will be contained from Paddle?

Reply:
@sshanzel You can find a reference in `PlusCard` and `PlusInfo` components (look for "extraLabel"), maybe we could create an utility component for handling this kind of labels (not mandatory) 👀

Reply:
Yeah, I think that would be amazing. Let me check if I can easily do that here.